File: Remove duplicates from an unsorted linked list[22-July-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution
{
    public:
    //Function to remove duplicates from unsorted linked list.
     Node* removeDuplicates(Node* head) {
        Node* ansList = new Node(-1);
        Node* ansptr = ansList;
        unordered_set<int> st;

        while (head) {
            if (st.find(head->data) == st.end()) {
                st.insert(head->data);
                ansptr->next = head;
                ansptr = ansptr->next;
            }

            head = head->next;
        }

        ansptr->next = nullptr;

        return ansList->next;
    }
};

File: Remove duplicates from an unsorted linked list[22-July-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    #Function to remove duplicates from unsorted linked list.
    def removeDuplicates(self, head):
        ansList=Node(-1)
        ansptr=ansList
        st=set()
        
        while(head):
            if(head.data not in st):
                st.add(head.data)
                ansptr.next=head
                ansptr=ansptr.next
            
            head=head.next

        ansptr.next=None
        
        return ansList.next
   